crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1618,7 +1621,7 @@ impl<'db> TypeInferenceBuilder<'db> {
 
                 let target_ty = enter_ty
                     .call(self.db(), &CallArguments::positional([context_expression_ty]))
-                    .return_type_result(&self.context, context_expression.into())
+                    .map(|outcome| outcome.return_type(self.db()))
```

Comment:
We probably want to have more fine granular error handlers here but that's out of the scope of this PR (and the same applies for other implicit call sites).

Reply:
Worth a TODO comment?